### PR TITLE
qat: rework device binding

### PR DESCRIPTION
**Pull request by mythi**
Closes: #715

### CODE REVIEW

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go [hunk 1]
```diff
@@ -93,11 +97,36 @@ func newDevicePlugin(pciDriverDir, pciDeviceDir string, maxDevices int, kernelVf
...
+func (dp *DevicePlugin) setupDeviceIDs() error {
+	for devID, driver := range qatDeviceDriver {
+		for _, enabledDriver := range dp.kernelVfDrivers {
+			if driver != enabledDriver {
+				continue
+			}
+
+			err := writeToDriver(filepath.Join(dp.pciDriverDir, dp.dpdkDriver, "new_id"), vendorPrefix+devID)
+			if err != nil && !errors.Is(err, os.ErrExist) {
+				return errors.Wrapf(err, "failed to set device ID %s for %s. Driver module not loaded?", devID, dp.dpdkDriver)
```
Comment (rojkov) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv.go:114]: nit: the error contains a trace stack already, no need to add one more
```suggestion
				return errors.WithMessagef(err, "failed to set device ID %s for %s. Driver module not loaded?", devID, dp.dpdkDriver)
```

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go [hunk 2]
```diff
@@ -197,34 +230,18 @@ func (dp *DevicePlugin) getDpdkMounts(dpdkDeviceName string) []pluginapi.Mount {
 	}
 }
 
-func (dp *DevicePlugin) getDeviceID(pciAddr string) (string, error) {
-	devID, err := os.ReadFile(filepath.Join(dp.pciDeviceDir, filepath.Clean(pciAddr), "device"))
+func getDeviceID(device string) (string, error) {
+	devID, err := os.ReadFile(filepath.Join(device, "device"))
 	if err != nil {
-		return "", errors.Wrapf(err, "Cannot obtain ID for the device %s", pciAddr)
+		return "", err
```
Comment (rojkov) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv.go:236]: Errors returned by external libs don't contain a trace stack, better have it wrapped.

Comment (mythi) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv.go:236]: I left it out here because it's just printed out as a warning.

Comment (rojkov) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv.go:236]: That's fine. I was trying to adhere the policy we documented in DEVEL.md.
Maybe a comment would be helpful here then.

**Review by rojkov — APPROVED** [review]

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go (hunk 2 above)
Comment (mythi) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv.go]: I decided to wrap to make it consistent with everything else.

**Review by rojkov — APPROVED** [review]
LGTM!

**Pull request merged**